test/unit/StakingProducts/recalculateEffectiveWeight.js

```diff
@@ -0,0 +1,98 @@
+const { ethers } = require('hardhat');
+const { expect } = require('chai');
+
+const { parseEther } = ethers.utils;
+const { Zero } = ethers.constants;
+
+const { allocateCapacity, depositTo, burnStake, setStakedProducts } = require('./helpers');
+
+const DEFAULT_PRODUCTS = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20];
+describe('recalculateEffectiveWeight', function () {
```

Comment:
other cases:

targetWeight is set high say 10

cover bought to make the activeWeight 8

targetWeight decreased to non-zero value

also test targetWeight decreased to 0 value